centreon/src/Core/Security/ProviderConfiguration/Domain/SecurityAccess/GroupsMapping.php

```diff
@@ -0,0 +1,183 @@
+<?php
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *  For more information : contact@centreon.com
+ */
+
+declare(strict_types=1);
+
+namespace Core\Security\ProviderConfiguration\Domain\SecurityAccess;
+
+use Centreon\Domain\Entity\ContactGroup;
+use Centreon\Domain\Log\LoggerTrait;
+use Core\Security\Authentication\Domain\Exception\AuthenticationConditionsException;
+use Core\Security\ProviderConfiguration\Domain\LoginLoggerInterface;
+use Core\Security\ProviderConfiguration\Domain\Model\Configuration;
+use Core\Security\ProviderConfiguration\Domain\OpenId\Model\ContactGroupRelation;
+use Core\Security\ProviderConfiguration\Domain\SecurityAccess\AttributePath\AttributePathFetcher;
+
+class GroupsMapping implements SecurityAccessInterface
+{
+
+    use LoggerTrait;
+
+    /**
+     * @var string
+     */
+    private string $scope;
```

Comment:
Not initialized in constructor : is this property useful ?
And if yes because it is part of the object "state", then please set a default value.
```suggestion
    private string $scope = '';
```